core-bundle/src/Security/Authentication/RememberMe/ExpiringTokenBasedRememberMeServices.php

```diff
@@ -0,0 +1,253 @@
+        }
+
+        return $rows;
+    }
+
+    private function migrateToken($token): string
+    {
+        $tokenValue = base64_encode(random_bytes(64));
+
+        $this->connection->update(
+            'tl_remember_me',
+            ['expires' => (new \DateTime())->add(new \DateInterval('PT60S'))],
+            ['expires' => DoctrineType::DATETIME]
+        );
+
+        $this->insertToken($token->class, $token->username, $token->series, $tokenValue);
+
+        return $tokenValue;
+    }
+
+    /**
+     * Adds a new rememberme token to the database.
+     */
+    private function insertToken(string $class, string $username, string $series, string $tokenValue): void
+    {
+        $this->connection->insert(
+            'tl_remember_me',
+            [
+                'class' => $class,
+                'username' => $username,
+                'series' => $series,
+                'value' => hash_hmac('sha256', $tokenValue, $this->secret),
+                'lastUsed' => new \DateTime(),
+            ],
+            [
+                'class' => \PDO::PARAM_STR,
+                'username' => \PDO::PARAM_STR,
+                'series' => \PDO::PARAM_STR,
+                'value' => \PDO::PARAM_STR,
+                'lastUsed' => DoctrineType::DATETIME,
+            ]
+        );
+    }
+
+    private function findValidToken(array $rows, string $tokenValue)
+    {
+        $tokenValue = hash_hmac('sha256', $tokenValue, $this->secret);
+
+        while ($token = array_shift($rows)) {
+            try {
+                if (!hash_equals($token->value, $tokenValue)) {
+                    throw new CookieTheftException(
+                        'This token was already used. The account is possibly compromised.'
+                    );
+                }
+
+                if ((new \DateTime($token->lastUsed))->getTimestamp() + $this->options['lifetime'] < time()) {
+                    throw new AuthenticationException('The cookie has expired.');
+                }
+
+                return $token;
+            } catch (AuthenticationException $e) {
+                $lastException = $e;
+            }
+        }
+
+        throw $lastException;
+    }
+
+    /**
+     * Creates a rememberme cookie.
+     */
+    private function createCookie(Request $request, string $series, string $tokenValue): Cookie
+    {
+        return new Cookie(
+            $this->options['name'],
+            $this->encodeCookie([$series, $tokenValue]),
+            time() + $this->options['lifetime'],
+            $this->options['path'],
+            $this->options['domain'],
+            $this->options['secure'] ?? $request->isSecure(),
+            $this->options['httponly'],
+            false,
+            $this->options['samesite'] ?? null
+        );
+    }
+
+    /**
+     * Clean up expired tokens on successful login (lazy behavior).
+     * This will also clean other series that have expired, e.g. if a cookie was never used again within the lifetime.
```

Comment:
Can you also add a comment about the `5 minutes` timespan here? That it seems to be a reasonable value. Just so that `new \DateInterval('PT300S')` does not look like a randomly chosen value if anybody looks into the code.